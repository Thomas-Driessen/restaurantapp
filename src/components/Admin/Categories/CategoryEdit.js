import React from 'react';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import TextField from "@material-ui/core/TextField";
import FormControl from "@material-ui/core/FormControl";
import { makeStyles } from "@material-ui/core/styles";
import clsx from "clsx";
import SaveIcon from "@material-ui/icons/Save";
import CloseIcon from "@material-ui/icons/Close";
import IconButton from "@material-ui/core/IconButton";
import EditIcon from '@material-ui/icons/Edit';

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
  },
  margin: {
    margin: theme.spacing(1),
  },
  withoutLabel: {
    marginTop: theme.spacing(2),
  },
  textField: {
    width: "50ch",
  }
}));

const Product = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const [values, setValues] = React.useState({ ...props.category });
  const [image, setImage] = React.useState("");
  const [loading, setLoading] = React.useState(false);

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  React.useEffect(() => {
    setValues(props.category);
  }, [props.category]);

  const uploadImage = async (e) => {
    const files = e.target.files;
    const data = new FormData();
    data.append("file", files[0]);
    data.append("upload_preset", `Categories`);
    setLoading(true);

    const res = await fetch(
      "https://api.cloudinary.com/v1_1/drb2yh2dy/image/upload",
      {
        method: "POST",
        body: data,
      }
    );

    const file = await res.json();

    setImage(file.secure_url);

    setLoading(false);
  };

  const editCategory = () => {
    let category = values;
    category.image = image ? image : category.image;

    fetch(`/api/category/${props.categoryType}`, {
      method: "PUT",
      mode: "cors",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify(category),
    }).catch(console.log);

    alert("Your changes have been saved");
    setOpen(false);
  }

  return (
    <div>
      <IconButton
        size="medium"
        color="primary"
        onClick={handleClickOpen}
      >
        <EditIcon />
      </IconButton>

      <Dialog open={open} onClose={handleClose} aria-labelledby="product-title">
        <DialogTitle id="product-title">
          {props.category.categoryName}
        </DialogTitle>
        <DialogContent>
          <CardMedia
            style={{ height: 400 }}
            component="img"
            height="250"
            src={props.category.image}
            alt={`Image for ${props.category.categoryName} Not Found`}
            title={props.category.categoryName}
          />
          {props.category.categoryName ? (
            <FormControl
              className={clsx(
                classes.margin,
                classes.withoutLabel,
                classes.textField
              )}
            >
              <TextField label="Title" value={values.categoryName} onChange={handleChange("categoryName")} />
            </FormControl>
          ) : null}
          <FormControl
            className={clsx(
              classes.margin,
              classes.withoutLabel,
              classes.textField
            )}
          >
            <input
              type="file"
              name="file"
              placeholder="Upload a file"
              onChange={uploadImage}
            />
          </FormControl>

          {loading ? (
            <DialogContentText>Loading...</DialogContentText>
          ) : null}
        </DialogContent>
        <DialogActions>
          <Button
            size="large"
            color="primary"
            target="_blank"
            onClick={editCategory}
            disabled={loading}
          >
            <SaveIcon /> Save
          </Button>
          <IconButton
            aria-label="close"
            color="primary"
            onClick={handleClose}
          >
            <CloseIcon />
          </IconButton>
        </DialogActions>
      </Dialog>
    </div>
  )
}
export default Product