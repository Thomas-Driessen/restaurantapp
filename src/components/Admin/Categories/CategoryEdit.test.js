import React from 'react';
import CategoryEdit from './CategoryEdit';
import renderer from 'react-test-renderer';

const category = {
  categoryName: "Cold"
};

it('CategoryEdit renders correctly', () => {
  const tree = renderer
    .create(<CategoryEdit category={category} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

afterAll(() => {
  global.gc && global.gc()
})