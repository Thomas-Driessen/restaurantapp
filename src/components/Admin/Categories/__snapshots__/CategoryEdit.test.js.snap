// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryEdit renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </span>
  </button>
</div>
`;
